### PR TITLE
fix: elliptic import

### PR DESCRIPTION
**Pull request by CodeSchwert**
### Hi👋, please prefix this PR's title with:

- [ ] `breaking-change:` if you have introduced modification that necessitates immediate adjustments by this SDK's users to their applications, clients, or integrations to avert disruptions to existing features or functionalities.
- [x] `feat:`, `fix:`, `refactor:`, `docs:`, or `chore:`.

# Summary

[DX-2760](https://immutable.atlassian.net/browse/DX-2760)

Proper fix for the `elliptic` package when using the SDK in ES module files.

# Detail and impact of the change
This change should not affect customers. It simply changes where x-client stark methods are being exported from internally. The external interface has not changed.

## Removed
- Core SDK as a dependency from x-client

## Fixed
- x-client dependency elliptic should work with ES module files

# Anything else worth calling out?

The `@types/elliptic` package was breaking the build.

[DX-2760]: https://immutable.atlassian.net/browse/DX-2760?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by carleeto — APPROVED** [review]

File: packages/x-client/tsconfig.json [hunk 1]
```diff
@@ -18,5 +18,5 @@
     "skipLibCheck": true
   },
   "include": ["src"],
-  "exclude": ["node_modules", "dist"]
+  "exclude": ["node_modules", "dist", "node_modules/elliptic"]
```
Comment (cajames) [inline packages/x-client/tsconfig.json:21]: Why is this required?

**Review by cajames — APPROVED** [review]

File: packages/x-client/tsconfig.json (hunk 1 above)
Comment (CodeSchwert) [reply packages/x-client/tsconfig.json:21]: Complains about missing types otherwise
